python/lsst/pipe/tasks/hips.py

```diff
@@ -0,0 +1,545 @@
+            )
+            qg.saveUri(args.save_qgraph)
+
+    @classmethod
+    def build_quantum_graph(
+        cls,
+        task_def,
+        registry,
+        constraint_order,
+        constraint_ranges,
+        where=None,
+        collections=None,
+    ):
+        """Generate a `QuantumGraph` for running just this task.
+
+        This is a temporary workaround for incomplete butler query support for
+        HEALPix dimensions.
+
+        Parameters
+        ----------
+        task_def : `lsst.pipe.base.TaskDef`
+            Task definition.
+        registry : `lsst.daf.butler.Registry`
+            Client for the butler database.  May be read-only.
+        constraint_order : `int`
+            HEALPix order used to contrain which quanta are generated, via
+            ``constraint_indices``.  This should be a coarser grid (smaller
+            order) than the order used for the task's quantum and output data
+            IDs, and ideally something between the spatial scale of a patch or
+            the data repository's "common skypix" system (usually ``htm7``).
+        constraint_ranges : `lsst.sphgeom.RangeSet`
+            RangeSet which describes constraint pixels (HEALPix NEST, with order
+            constraint_order) to constrain generated quanta.
+        where : `str`, optional
+            A boolean `str` expression of the form accepted by
+            `Registry.queryDatasets` to constrain input datasets.  This may
+            contain a constraint on tracts, patches, or bands, but not HEALPix
+            indices.  Constraints on tracts and patches should usually be
+            unnecessary, however - existing coadds that overlap the given
+            HEALpix indices will be selected without such a constraint, and
+            providing one may reject some that should normally be included.
+        collections : `str` or `Iterable` [ `str` ], optional
+            Collection or collections to search for input datasets, in order.
+            If not provided, ``registry.defaults.collections`` will be
+            searched.
+        """
+        config = task_def.config
+
+        dataset_types = pipeBase.PipelineDatasetTypes.fromPipeline(pipeline=[task_def], registry=registry)
+        # Since we know this is the only task in the pipeline, we know there
+        # is only one overall input and one overall output.
+        (input_dataset_type,) = dataset_types.inputs
+
+        # Extract the main output dataset type (which needs multiple
+        # DatasetRefs, and tells us the output HPX level), and make a set of
+        # what remains for more mechanical handling later.
+        output_dataset_type = dataset_types.outputs[task_def.connections.hips_exposures.name]
+        incidental_output_dataset_types = dataset_types.outputs.copy()
+        incidental_output_dataset_types.remove(output_dataset_type)
+        (hpx_output_dimension,) = (d for d in output_dataset_type.dimensions if d.name != "band")
```

Comment:
I think this is a bit fragile, what if there are more dimensions?

Reply:
@TallJimbo ?

Reply:
It's either this or a loop with `isinstance(SkyPixDimension, ...)` checks; figured this was slightly better (it's far from the only thing that would break if the dimensions change, and I consider that unlikely).

Reply:
what I mean is what if the dimensions are `("visit", "detector", "band", "healpix9")` then this will be too large to unpack, could it be switched to be a matching check instead of a not match check?

Reply:
I don't understand that, this is meant to be run on coadds.

Reply:
I just used it as an example, it might be any set of dimensions, though I might not be following what Jim is doing here

Reply:
My translation of what Nate is suggesting to concrete code is:
```
(hpx_output_dimension,) = (d for d in output_dataset_type.dimensions if isinstance(d, SkyPixDimension))
```
That does express what we're trying to do a bit more clearly, which I think is a better argument in this case than fragility (this whole function will be invalidated by those dimensions changing).  Whether it's better overall depends on how much you dislike `isinstance`.  This also gets a :man_shrugging: from me.